container/src/utilities/helpers.ts

```diff
@@ -21,22 +21,32 @@ export class GenericHelpersClass {
 
   /**
    * Checks whether web component is an attribute or property. In case of attribute, it returns the parsed value.
-   * @param webcomponent 
-   * @returns returns the parsed webcomponent value.
+   * @param {object | boolean | string} webcomponent value can either be an object, boolean or a stringified object, e.g webcomponent='{"selfregistered":"true"}'
+   * @returns {object | boolean} webcomponent returns the parsed webcomponent value.
    */
-  checkWebcomponentValue(webcomponent: any): object | boolean {
+  checkWebcomponentValue(webcomponent: object | boolean | string): object | boolean {
     if (typeof webcomponent === 'string') {
-      let parsedValue = JSON.parse(webcomponent);
-      if (this.isObject(parsedValue)) {
-        if ('selfRegistered' in parsedValue) {
-          parsedValue.selfRegistered = parsedValue.selfRegistered === 'true';
-        }
-      }
-      return parsedValue;
+      return JSON.parse(webcomponent);
```

Comment:
I would add the same comment here about handling it same as in **resolveContext** with a try/catch

Reply:
see _context_